multicluster/controllers/multicluster/commonarea/remote_common_area.go

```diff
@@ -206,29 +207,21 @@ func getSecretCACrtAndToken(secretObj *v1.Secret) ([]byte, []byte, error) {
 }
 
 func (r *remoteCommonArea) SendMemberAnnounce() error {
-	memberAnnounceList := &multiclusterv1alpha1.MemberClusterAnnounceList{}
-	if err := r.List(context.TODO(), memberAnnounceList, client.InNamespace(r.GetNamespace())); err != nil {
+	var err error
+	memberAnnounceName := "member-announce-from-" + r.GetLocalClusterID()
+	existingMemberAnnounce := &multiclusterv1alpha1.MemberClusterAnnounce{}
+	if err = r.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.GetNamespace(),
+		Name:      memberAnnounceName,
+	}, existingMemberAnnounce); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-	var localClusterMemberAnnounce multiclusterv1alpha1.MemberClusterAnnounce
-	localClusterMemberAnnounceExist := false
-	if len(memberAnnounceList.Items) != 0 {
-		for _, memberAnnounce := range memberAnnounceList.Items {
-			if memberAnnounce.ClusterID == r.GetLocalClusterID() {
-				localClusterMemberAnnounceExist = true
-				localClusterMemberAnnounce = memberAnnounce
-				break
-			}
-		}
-	}
-	if localClusterMemberAnnounceExist {
-		localClusterMemberAnnounce.LeaderClusterID = ""
-		leaderID := r.ClusterID
-		r.updateLeaderStatus(leaderID)
-		if leaderID != common.InvalidClusterID {
-			localClusterMemberAnnounce.LeaderClusterID = string(leaderID)
-		}
 
+	localClusterMemberAnnounceExists := !apierrors.IsNotFound(err)
```

Comment:
Is it safe to pass nil to apierrors.IsNotFound()? I would suggest:

localClusterMemberAnnounceExists := err == nil

Reply:
Done